plugins/contextmenu/plugin.js

```diff
@@ -59,8 +59,15 @@ CKEDITOR.plugins.add( 'contextmenu', {
 								// which make this property unreliable. (https://dev.ckeditor.com/ticket/4826)
 								( CKEDITOR.env.webkit ? holdCtrlKey : ( CKEDITOR.env.mac ? domEvent.$.metaKey : domEvent.$.ctrlKey ) );
 
-						if ( nativeContextMenuOnCtrl && isCtrlKeyDown ) {
+						if ( contextMenuSwitchingEnabled ) {
 							return;
+							if ( nativeContextMenuOnByDefault && !isCtrlKeyDown ) {
+								return;
+							}
+							
+							if ( !nativeContextMenuOnByDefault  && isCtrlKeyDown ) {
+								return;
+							}
```

Comment:
This is basically so called "dead code" - the `return` above will stop the execution so these `if`s are never reached. Which means it doesn't do anything :thinking: